viper/parser/stmt.py

```diff
@@ -185,6 +191,83 @@ def parse_for(self):
         self.context.forvars[varname] = True
         return o
 
+    def _is_list_iter(self):
+        """
+        Test if the current statement is a type of list, used in for loops.
+        """
+
+        # Check for literal or memory list.
+        iter_var_type = self.context.vars.get(self.stmt.iter.id).typ if isinstance(self.stmt.iter, ast.Name) else None
+        if isinstance(self.stmt.iter, ast.List) or isinstance(iter_var_type, ListType):
+            return True
+
+        # Check for storage list.
+        if isinstance(self.stmt.iter, ast.Attribute):
+            iter_var_type = self.context.globals.get(self.stmt.iter.attr)
+            if iter_var_type and isinstance(iter_var_type.typ, ListType):
+                return True
+
+        return False
+
+    def parse_for_list(self):
+        from .parser import (
+            parse_body,
+            make_setter
+        )
```

Comment:
Bad form to have imports inside a function, as that import will be executed every time the function is called

Reply:
I agree 100%, it needs to be done now but as we refactor `parser.py` I think it'll be easier to remove the imports inside functions.